### PR TITLE
DomainTools - Set Key to support creds

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] Ready

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

File: Packs/DomainTools/ReleaseNotes/1_1_1.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### DomainTools
+- Added the *Key* integration parameter to support credentials fetching object.
```
Comment (dantavori) [inline Packs/DomainTools/ReleaseNotes/1_1_1.md:4]:
```suggestion
- Added the *API Key* integration parameter to support credentials fetching object.
```

File: Packs/DomainTools/Integrations/DomainTools/DomainTools.yml [hunk 2]
```diff
@@ -34,6 +34,11 @@ configuration:
   defaultvalue: ""
   type: 8
   required: false
+- display: API Key (Deprecated)
+  name: key
```
Comment (dantavori) [inline Packs/DomainTools/Integrations/DomainTools/DomainTools.yml:38]:
```suggestion
- display: API Key (Deprecated)
  hidden: true
  name: key
```

File: Packs/DomainTools/Integrations/DomainTools/DomainTools.js [hunk 3]
```diff
@@ -300,27 +300,31 @@ var callWhoisHistory = function(url, domain){
 }
 
 var url = params.server.replace(/[\/]+$/, '');
+var key = (params.key || params.credentials.password);
+if (!key) {
+    log('API key must be provided.')
```
Comment (dantavori) [inline Packs/DomainTools/Integrations/DomainTools/DomainTools.js:305]:
```suggestion
    throw 'API key must be provided.'
```

File: Packs/DomainTools/Integrations/DomainTools/DomainTools.js [hunk 4]
```diff
@@ -300,27 +300,31 @@ var callWhoisHistory = function(url, domain){
 }
 
 var url = params.server.replace(/[\/]+$/, '');
+var key = (params.key || params.credentials.password);
```
Comment (dantavori) [inline Packs/DomainTools/Integrations/DomainTools/DomainTools.js:303]:
```suggestion
params.key = params.key || params.credentials.password
```
then you don't need to add key as an argument everywhere

**Review by dantavori — APPROVED** [review]
you can merge when completing the small fixes

File: Packs/DomainTools/Integrations/DomainTools/DomainTools.js (hunk 4 above)
Comment (yaakovpraisler) [reply Packs/DomainTools/Integrations/DomainTools/DomainTools.js]: done

**Pull request merged**